File: 3442-maximum-difference-between-even-and-odd-frequency-i/3442-maximum-difference-between-even-and-odd-frequency-i.cpp
class Solution {
public:
    int maxDifference(string s) {
        unordered_map<char,int> freq;
        for(auto ch:s) {
            freq[ch]++;
        }
        int minEven = s.length();
        int maxEven = 0;
        int minOdd = s.length();
        int maxOdd = 0;
        for(auto [_,count]:freq) {
            if(count%2==0) {
                minEven = min(minEven,count);
                maxEven = max(maxEven,count);
            } else {
                minOdd = min(minOdd,count);
                maxOdd = max(maxOdd,count);
            }
        }
        return max(maxEven-minOdd,maxOdd-minEven);
    }
};

File: 3442-maximum-difference-between-even-and-odd-frequency-i/3442-maximum-difference-between-even-and-odd-frequency-i.java
class Solution {
    public int maxDifference(String s) {
        int n = s.length();
        int maxOdd=0;;
        int minEven=n;
        int[] freq = new int[26];
        for(int i=0;i<n;i++) {
            freq[s.charAt(i)-'a']++;
        }

        for(int i=0;i<26;i++) {
            int count = freq[i];
            if(count==0) continue;
            if(count%2==0) {
                minEven = Math.min(minEven,count);
            } else {
                maxOdd = Math.max(maxOdd,count);
            }
        }
        return maxOdd-minEven;
    }
}